docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,492 @@
+# MeshExternalService API
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/6330
+
+## Context and Problem Statement
+
+The current implementation of `ExternalService` has significant limitations and warrants a complete redesign.
+
+* Lack of support for * subdomains
+* Lack of support for thick clients (think Kafka or C* where discovery is done as part of the client).
+* Need for pluggability (for example lambda support or equivalent)
+* Working well with gateways (what if your ExternalService relies on SNI/host header to route through a gateway)
+* Applying policies with and without egress
+
+## Considered Options
```

Comment:
I think we shouldn't remove considered options - 

![image](https://github.com/kumahq/kuma/assets/2753650/dbc45552-6387-49bc-a78c-bab7c1953b43)

that way in the future if we go back to this we can go - yeah we thought about this and decided not to go that way